File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        if(nums.size() < 2) return nums.size();
        
        int left = 2;
        int right = 2;
        
        while(right < nums.size()) {
            
            if(nums[left - 2] != nums[right]) {
                // the right position
                nums[left] = nums[right];
                left++;
                right++;
            }
            else {
                // find the right position
                right++;
            }
            
        }
        
        return left;
    }
};

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        
        int i=0 , j=0 , count = 0;
        int limit = 2;
        
        while(j < nums.length) {
            int current = nums[j];
            count = 0;
            
            while(j < nums.length && nums[j] == current) {
                count++;
                j++;
            }
            
            int canTake = Math.min(limit , count);
            
            // to fill up in between positions
            for(int k=0 ; k<canTake ; k++) {
                nums[i] = current;
                i++;
            }
        }
        
        return i;
    }
}